HW04.Rmd

```diff
@@ -1,126 +1,266 @@
+    labs(x = "Month", y = "Gun death per month", title = "Gun death per month")+
+    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
+        panel.background = element_blank(), axis.line = element_line(colour = "black"), 
+        legend.position = "none")
+
+```
+
+## Generating a bar chart that identifies the number of gun deaths associated with each type of intent cause of death. The bars should be sorted from highest to lowest values.
+
+```{r intent}
+
+gun_intent <- gun$intent
+gun_intent <- as.data.frame(table(gun_intent))
+colnames(gun_intent) <- c("intent","death")
+
+ggplot(gun_intent)+
+    geom_bar(aes(x = intent,  y = death, fill = intent), stat = "identity")+
+    labs(x = "Intent", y = "Gun death", title = "Gun death for each intent")+
+    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
+        panel.background = element_blank(), axis.line = element_line(colour = "black"), 
+        legend.position = "none")
+
+```
+
+## Generating a boxplot visualizing the age of gun death victims, by sex. Print the average age of female gun death victims.
+
+```{r age_&_sex}
+
+gun_age_sex <- gun[,5:6]
+#gun_age_sex <- as.data.frame(table(gun_age_sex))
+#colnames(gun_age_sex) <- c("sex", "age", "death")
+
+#remove the not available line
+gun_age_sex <- subset(gun_age_sex, age != "Not Available")
+#gun_age_sex <- subset(gun_age_sex, death != 0)
+gun_age_sex$age <- as.numeric(gun_age_sex$age)
+
+#draw the plot
+ggplot(gun_age_sex)+
+    geom_boxplot(aes(x = sex, y = age, fill = sex))
+
+#calculate average female age
+gun_age_F <- subset(gun_age_sex, sex == "F")
+mean(gun_age_F$age)
+
+```
+
+
+## How many white males with at least a high school education were killed by guns in 2012?
+
+```{r education}
+
+gun_white_2012 <- subset(gun, race == "White" & year == "2012")
+gun_white_2012 <- subset(gun_white_2012, education != "Not Available" & education != "Less than HS")
+#check to make sure all less HS are filtered out
+table(gun_white_2012$education)
+#print out number of white males with at least a high school education were killed by guns in 2012
+nrow(gun_white_2012)
+
+```
+
+
+## Which season of the year has the most gun deaths?
+
+Assume that:
+
+* Winter = January-March
+* Spring = April-June
+* Summer = July-September
+* Fall = October-December
+
+```{r season}
+gun$month[gun$month == "1"] <- "Winter"
+gun$month[gun$month == "2"] <- "Winter"
+gun$month[gun$month == "3"] <- "Winter"
+gun$month[gun$month == "4"] <- "Spring"
+gun$month[gun$month == "5"] <- "Spring"
+gun$month[gun$month == "6"] <- "Spring"
+gun$month[gun$month == "7"] <- "Summer"
+gun$month[gun$month == "8"] <- "Summer"
+gun$month[gun$month == "9"] <- "Summer"
+gun$month[gun$month == "10"] <- "Fall"
+gun$month[gun$month == "11"] <- "Fall"
+gun$month[gun$month == "12"] <- "Fall"
+#is there faster way to do this?, instead of type out all of them?
```

Comment:
To do this quicker, I wonder if you could do: gun$month[gun$month == "1:3"] <- "Winter"     ????